src/control/lib/control/system.go

```diff
@@ -102,28 +150,43 @@ func SystemReformat(ctx context.Context, rpcClient UnaryInvoker, req *SystemRefo
 		return nil, err
 	}
 
-	resp := new(SystemReformatResp)
-	if err = convertMSResponse(ur, resp); err != nil {
-		return nil, err
+	// MS response will contain collated results for all ranks
+	resetResp := new(SystemResetFormatResp)
+	if err = convertMSResponse(ur, resetResp); err != nil {
+		return nil, errors.WithMessage(err, "converting MS to reformat resp")
 	}
 
-	if resp.Results.HasErrors() {
-		return resp, nil // return early if any errors from SystemResetFormat
+	resetRankErrors, hostList, err := getResetRankErrors(resetResp.Results)
+	if err != nil {
+		return nil, err
 	}
 
 	// ranks in AwaitFormat state, trigger format
+	// TODO: remove failed ranks from StorageFormat() rank list
 	formatReq := &StorageFormatReq{Reformat: true}
-	formatReq.SetHostList(resp.Hosts)
+	formatReq.SetHostList(hostList)
```

Comment:
Hmm. I think there may have been some miscommunication on this point. I don't think it makes sense to proceed to format any hosts if the prep stage fails for any rank. What I was trying to express with my previous feedback is that the SystemFormatResp struct shouldn't expose the raw rank results and require API callers to check a separate structure. This commit is moving in the right direction, in that failed prep results are converted into HostErrors, but I think the conversion should happen before proceeding to format and the call should exit early when prep fails.